src/main/java/com/contentgrid/gateway/security/jwt/issuer/PropertiesBasedJwtClaimsSigner.java

```diff
@@ -121,9 +98,26 @@ public SignedJWT sign(JWTClaimsSet jwtClaimsSet) throws JOSEException {
             signedJwt.sign(selectedSigner);
             return signedJwt;
         }
-        throw new IllegalStateException("No active signing keys support any of the configured algorithms (%s); algorithms that can be used by these keys are %s".formatted(
-                properties.getAlgorithms(),
-                algorithmsSupportedByKeys
-        ));
+        throw new IllegalStateException(
+                "No active signing keys support any of the configured algorithms (%s); algorithms that can be used by these keys are %s".formatted(
+                        algorithms,
+                        algorithmsSupportedByKeys
+                ));
+    }
+
+    private ConcurrentLruCache<JWK, JWSSigner> signerCache;
+
+    private JWSSigner getJwsSigner(JWK jwk) {
+        if (signerCache == null) {
+            signerCache = new ConcurrentLruCache<>(100,
+                    key -> {
+                        try {
+                            return jwsSignerFactory.createJWSSigner(key);
+                        } catch (JOSEException e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+        }
+        return signerCache.get(jwk);
```

Comment:
1. I think this can be moved to the constructor?
2. A capacity of 100 is maybe a bit large when you expect typically 1 active signing key; maybe sometimes 2 or 3 (for different key types). No JWSSigners are created for retired keys.